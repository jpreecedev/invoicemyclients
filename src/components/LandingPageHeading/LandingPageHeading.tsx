import React from 'react'
import { Button, Container, Header, Icon } from 'semantic-ui-react'
import { Link } from 'react-router-dom'

import { siteConfig } from '../../site-config'
import { ROUTES } from '../../routes'

interface LandingPageHeadingProps {
  mobile?: boolean
}

const LandingPageHeading: React.FC<LandingPageHeadingProps> = ({ mobile = false }) => (
  <Container text>
    <Header
      as="h1"
      content={siteConfig.title}
      inverted
      style={{
        fontSize: mobile ? '2em' : '4em',
        marginBottom: 0,
        marginTop: mobile ? '1.5em' : '3em'
      }}
    />
    <Header
      as="h2"
      content={siteConfig.tagline}
      inverted
      style={{
        fontSize: mobile ? '1.5em' : '1.7em',
        marginBottom: '2rem'
      }}
    />
    <Button primary size="huge" as={Link} to={ROUTES.SignUpPage}>
      It&apos;s free to get started
      <Icon name="arrow right" />
    </Button>
  </Container>
)

export { LandingPageHeading }
