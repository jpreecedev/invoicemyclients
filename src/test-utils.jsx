import React from 'react'
import { MemoryRouter as Router } from 'react-router-dom'
import { render } from '@testing-library/react'
import { createMemoryHistory } from 'history'

function renderWithRouter(
  ui,
  { route = '/', history = createMemoryHistory({ initialEntries: [route] }) } = {}
) {
  const Wrapper = ({ children }) => <Router history={history}>{children}</Router>
  return {
    ...render(ui, { wrapper: Wrapper }),
    // adding `history` to the returned utilities to allow us
    // to reference it in our tests (just try to avoid using
    // this to test implementation details).
    history
  }
}

export { renderWithRouter }
