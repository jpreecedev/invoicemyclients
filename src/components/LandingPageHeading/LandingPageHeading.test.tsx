import React from 'react'

import { renderWithRouter } from '../../test-utils'
import { LandingPageHeading } from './LandingPageHeading'

describe('<LandingPageHeading /> tests', () => {
  it('should match mobile snapshot', () => {
    expect(renderWithRouter(<LandingPageHeading mobile={false} />).container).toMatchSnapshot()
  })

  it('should match desktop snapshot', () => {
    expect(renderWithRouter(<LandingPageHeading mobile={true} />).container).toMatchSnapshot()
  })
})
