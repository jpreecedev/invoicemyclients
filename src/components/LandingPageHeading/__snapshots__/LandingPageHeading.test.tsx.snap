// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LandingPageHeading /> tests should match desktop snapshot 1`] = `
<div>
  <div
    class="ui text container"
  >
    <h1
      class="ui inverted header"
      style="font-size: 2em; margin-bottom: 0px; margin-top: 1.5em;"
    >
      InvoiceMyClients.com
    </h1>
    <h2
      class="ui inverted header"
      style="font-size: 1.5em; margin-bottom: 2rem;"
    >
      We make it quick, easy, and secure for your clients to pay your invoices.
    </h2>
    <a
      class="ui huge primary button"
      href="/sign-up"
      role="button"
    >
      It's free to get started
      <i
        aria-hidden="true"
        class="arrow right icon"
      />
    </a>
  </div>
</div>
`;

exports[`<LandingPageHeading /> tests should match mobile snapshot 1`] = `
<div>
  <div
    class="ui text container"
  >
    <h1
      class="ui inverted header"
      style="font-size: 4em; margin-bottom: 0px; margin-top: 3em;"
    >
      InvoiceMyClients.com
    </h1>
    <h2
      class="ui inverted header"
      style="font-size: 1.7em; margin-bottom: 2rem;"
    >
      We make it quick, easy, and secure for your clients to pay your invoices.
    </h2>
    <a
      class="ui huge primary button"
      href="/sign-up"
      role="button"
    >
      It's free to get started
      <i
        aria-hidden="true"
        class="arrow right icon"
      />
    </a>
  </div>
</div>
`;
